app/_src/gateway/upgrade/backup-and-restore.md

```diff
@@ -0,0 +1,270 @@
+
+{% endnavtab %}
+{% endnavtabs %}
+
+## Restore Gateway entities
+
+### Database-native restore
+
+To recover {{site.base_gateway}} configuration data from a database-native backup, make sure the database is prepared first.
+
+For PostgreSQL:
+
+1. In `kong.conf`, set a database user using the `pg_user` parameter:
+
+    ```
+    pg_user = kong
+    ```
+
+2. In `kong.conf`, set a database name using the `pg_database` parameter:
+
+    ```
+    pg_database = kong
+    ```
+
+3. Bootstrap database entities using the `migrations` command. 
+Refer to the [`kong migrations` CLI reference](/gateway/{{page.kong_version}}/reference/cli/#kong-migrations) 
+for more information.
+
+    ```sh
+    kong migrations bootstrap
+    ```
+
+4. You can now restore the data using the utility `pg_restore`:
+
+    ```sh
+    pg_restore -U kong -C -d postgres --if-exists --clean kongdb_backup_20230816/
+    ```
+
+### Declarative restore
+
+If you need to roll back, change the {{site.base_gateway}} instance back to the original version, 
+validate the declarative config, then apply it to your {{site.base_gateway}} instance.
+
+{% navtabs %}
+{% navtab Traditional or hybrid mode - decK %}
+
+In traditional or hybrid mode, use decK to restore your configuration from a backup state file.
+
+1. Check that {{site.base_gateway}} is online:
+
+    ```sh
+    deck ping
+    ```
+2. Validate the declarative config:
+
+    ```sh
+    deck validate [--online] -s /path/to/kong_backup.yaml
+    ```
+
+3. Once verified, restore a particular workspace or all workspaces at once:
+
+    ```sh
+    deck sync --all-workspaces -s /path/to/kong_backup.yaml
+    ```
+    or
+
+    ```sh
+    deck sync --workspace it_dept -s /path/to/kong_backup.yaml
+    ```
+
+{% endnavtab %}
+{% navtab Traditional or hybrid mode - Kong CLI %}
+
+If you backed up {{site.base_gateway}} database using `kong config db_export`, 
+use the Kong CLI to restore your configuration from the backup declarative config file.
+
+1. Validate the backup configuration file before restoring it:
+
+    ```sh
+    kong config parse /path/to/kong_backup.yaml
+    ```
+
+2. Import entities into your database:
+
+    ```sh
+    kong config db_import /path/to/kong_backup.yaml
+    ```
+
+2. Restart or reload your {{site.base_gateway}} instance:
+
+    ```sh
+    kong restart
+    ```
+
+    or
+
+    ```sh
+    kong reload
+    ```
+
+{% endnavtab %}
+{% navtab DB-less mode %}
+
+In DB-less mode, use the Kong CLI to restore your configuration from a declarative config file.
+
+1. Validate the backup configuration file before restoring it:
+
+    ```sh
+    kong config parse /path/to/kong_backup.yaml
+    ```
+
+2. Restart or reload your {{site.base_gateway}} instance using the backup configuration file:
+
+    ```sh
+    kong restart -c /path/to/kong_backup.yaml
+    ```
+
+    or
+
+    ```sh
+    kong reload -c /path/to/kong_backup.yaml
+    ```
```

Comment:
This is NOT right! To restore with the declarative backup file, please point [declarative_config](https://docs.konghq.com/gateway/latest/reference/configuration/#declarative_config) in `kong.conf` to `/path/to/kong_backup.yaml`.

```suggestion
2. Restart or reload your {{site.base_gateway}} instance using the backup configuration file:

    ```sh
    export KONG_DECLARATIVE_CONFIG=kong.yml; kong restart -c /path/to/kong_backup.yaml
    # -or-
    export KONG_DECLARATIVE_CONFIG=kong.yml; kong reload -c /path/to/kong_backup.yaml
    ```

    Alternatively, post the declarative backup file to `:8001/config` endpoint.

    ```sh
    curl -sS http://localhost:8001/config?check_hash=1 -F 'config=@kong_backup.yaml' ; echo
    ```
```